search/roadmap.adoc

```diff
@@ -26,34 +26,46 @@ Analyzers::
 A new DSL will be introduced to allow people to define Analyzers programmatically at boostrap, as an alternative to the current `@Analyzer` and `@AnalyzerDef` annotations.
 We might want to automatically register a set of commonly used Analyzers - let us know what you think of this as the choice is hard!
 
+Java 9 compatibibility::
+We're aiming at regularly testing compatibility with Java 9 on top of the regular tests on Java 8.
+Testing will be limited to having Hibernate Search on the regular classpath: running as a module is an additional step we deferred to version 5.9.
+
+
+== Hibernate Search 5.9
+
+Elasticsearch performance::
+We applied some interesting performance improvements in version 5.8 but there are some more interesting ideas which could provide
+a radical boost. Moved these to 5.9 as they are quite complex and need fleshing out.
+
 WildFly Swarm::
-WildFly Swarm is the new fine-grained modular version of WildFly aiming at microservices.
+http://wildfly-swarm.io/[WildFly Swarm] is the new fine-grained modular version of http://wildfly.org/[WildFly] aiming at microservices.
 We should make sure Hibernate Search is integrated properly and the usage is documented.
 
-Java 9 compatibibility::
-We're aiming at regularly testing compatibility with Java 8 and now 9 too.
-Hibernate Search should test its modularity within project Jigsaw, within reason as we can't depent on Java9 release schedules.
-
 Removal of class definitions from master nodes::
 The integration of a "master node" needs to be simplified, both in terms of user configuration and implementation.
 We plan to not need user classes (entities) on such a node, so that in future it could be a separate download
 which works out of the box with minimal configuration.
 
+Modularity related enhancements::
+Some code needs to be moved to different jars to improve the experience with OSGi, JBoss Modules and eventually Java 9 (Jigsaw) users.
```

Comment:
Good idea, but can we do that in a minor? I doubt we can make it transparent to users, since we'd need to make the engine depend on the new JAR, and that's not possible because it would introduce a cyclic dependency.

Reply:
This is really just a way to describe my local Tika refactoring branch. I decided to not do it in 5.8 as we're in CR1 already but it's almost done, would be good to integrate it.

It can be transparent to the users in terms of Java source code, they might have to add a dependency but that's typically ok. Dependencies change all the time in minors, typically it's just version numbers but the point is people still have to shuffle jar files around and amend deployment scripts. Also remember: they won't need to do anything if they don't use Tika. If they were using Tika in such a modular environment .. it wasn't working to begin with so you might consider it a new feature ;)